src/utils/math.ts

```
@@ -20,7 +23,48 @@
+    upperboundValue: T,
+): Nullable<number> => {
+    if (highIdx < lowIdx || "undefined" === typeof (get(highIdx))) {
+        return null;
+    }
+
+    while (lowIdx <= highIdx) {
+        const mid = Math.floor((lowIdx + highIdx) / 2);
+
+        // `mid` is guaranteed to be within bounds since `low <= high`.
+        if (get(mid) <= upperboundValue) {
+            lowIdx = mid + 1;
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)